plugin/src/main/java/io/fabric8/maven/plugin/BuildMojo.java

```diff
@@ -241,7 +293,191 @@ private void executeOpenShiftBuild(ServiceHub hub, ImageConfiguration imageConfi
+        }
+        TagReference tag = null;
+        if (tags.isEmpty()) {
+            tag = new TagReferenceBuilder().build();
+            tags.add(tag);
+        } else {
+            tag = tags.get(tags.size() - 1);
+        }
+        ObjectReference from = tag.getFrom();
+        if (from == null) {
+            from = new ObjectReference();
+            tag.setFrom(from);
+        }
+
+        boolean answer = false;
+        if (!Objects.equals(label, tag.getName())) {
+            tag.setName(label);
+            answer = true;
+        }
+        if (!Objects.equals(kind, from.getKind())) {
+            from.setKind(kind);
+            answer = true;
+        }
+        if (!Objects.equals(namespace, from.getNamespace())) {
+            from.setNamespace(namespace);
+            answer = true;
+        }
+        if (!Objects.equals(name, from.getName())) {
+            from.setName(name);
+            answer = true;
+        }
+        if (answer) {
+            log.info("Updated ImageStream " + buildConfigName + " to namespace: " + namespace + " name: " + name);
+        }
+        return answer;
+    }
+
+    private static String getImageLabel(String imageName) throws MojoExecutionException {
+        int idx = imageName.lastIndexOf(':');
+        if (idx < 0) {
+            throw new MojoExecutionException("No ':' in the image name:  " + imageName);
+        } else {
+            return imageName.substring(idx + 1);
+        }
+    }
+
+
+    private String findTagSha(ImageStream imageStream) throws MojoExecutionException {
+        ImageStreamStatus status = imageStream.getStatus();
+        if (status != null) {
+            List<NamedTagEventList> tags = status.getTags();
+            if (tags != null && !tags.isEmpty()) {
+                // latest tag is the first
+                for (NamedTagEventList list : tags) {
+                    List<TagEvent> items = list.getItems();
+                    if (items != null) {
+                        // latest item is the first
+                        for (TagEvent item : items) {
+                            String image = item.getImage();
+                            if (Strings.isNotBlank(image)) {
+                                return image;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        throw new MojoExecutionException("Could not find a tag in the ImageStream " + KubernetesHelper.getName(imageStream));
+    }
+
+    private boolean isBuildCompleted(Watcher.Action action, Build build) {
+        BuildStatus buildStatus = build.getStatus();
+        if (buildStatus != null) {
+            String status = buildStatus.getPhase();
+            if (Strings.isNotBlank(status)) {
+                if (!Objects.equals(status, lastBuildStatus)) {
+                    lastBuildStatus = status;
+                    log.info("Build " + getName(build) + " status: " + status);
```

Comment:
Also recommend to use the "%s" form as it can bork if the status or name contains characters building an invalid format.

E.g.

```java
log.verbose("Build %s : %s",getName(build),status);
```